### ISSUE TITLE
plugin seems to be slowly filling up process table with <defunct>

### ISSUE THREAD

**Issue by eleaner** [kadimasolutions/docker-plugin_lizardfs#9]
Hi,

my VPSes are dying because there are not enough processes.
I traced it back to thousands of processes like that
root     31304  1149  0 12:30 ?        00:00:00 [mfsmount] <defunct>
the father 1149 is
root      1149  1096  0 12:22 ?        00:00:01 node index.js

I am pretty sure it is the lizardfs plugin

I know too little to understand what's going on. Could you please help?
I could periodically reset my VPSes but it does not sound like the best option.

**Comment by zicklag** [comment]
Oh, that isn't good.

It must be either failing to unmount the volumes or, for some reason, the `mfsmount` process isn't exiting when it does the `unmount` ( right [here](https://github.com/katharostech/docker-plugin_lizardfs/blob/master/index.js#L356) in the code ).

Is there any kind of "abnormal" behavior that you might know of in your particular use-case? Like creating and removing containers with mounts frequently or something like that, or is it mostly long-running containers that don't change much?

Unfortunately I'm not sure exactly when I'll have time to look into this, but I'll try to look at it if I get the chance.

**Comment by eleaner** [comment]
it feels like any time a container/swarm service dies because of whatever reason, the mfsmount process of the used volume stays. Possibly because it is not a child of the dying service but of the plugin that is still alive.
due to some problems, one of my services kept dying every few seconds
all seems to be good when I simply bind container/service to mounted lizard

**Comment by zicklag** [comment]
Yeah, I'll have to add some log messages in there and test killing containers to see what is happening in the plugin when a container dies. It sounds like it should be fixable.
